pkg/private/pkg_files.bzl

```diff
@@ -449,6 +449,41 @@ def add_from_default_info(
                     gid = mapping_context.default_gid,
                 )
 
+            # if bzlmod is enabled, create _repo_mapping under runfiles directory
+            if (
+                getattr(src[DefaultInfo], "files_to_run") and
+                hasattr(src[DefaultInfo].files_to_run, "repo_mapping_manifest") and
+                src[DefaultInfo].files_to_run.repo_mapping_manifest != None
+            ):
+                repo_mapping_manifest = src[DefaultInfo].files_to_run.repo_mapping_manifest
+                mapping_context.file_deps.append(depset([repo_mapping_manifest]))
+
+                d_path = mapping_context.path_mapper(dest_path(
+                    repo_mapping_manifest,
+                    data_path,
+                    data_path_without_prefix,
+                ))
+                add_single_file(
+                    mapping_context,
+                    dest_path = d_path,
+                    src = repo_mapping_manifest,
+                    origin = src.label,
+                    mode = mapping_context.default_mode,
+                    user = mapping_context.default_user,
+                    group = mapping_context.default_group,
+                )
```

Comment:
This should really be a symlink into `.runfiles/_repo_mapping`, but I have no idea how to add a symlink that also respect `remap_paths` here.

Reply:
Can you add that comment to the code as a TODO, so some future maintainer sees it.

Reply:
Done in abdf4ae